tests/conftest.py

```diff
@@ -200,331 +178,97 @@ def dummy_input_bundle():
-        except web3.exceptions.NoABIFunctionsFound:
-            fn_names = []
-
-        for fn_name in fn_names:
-            # Override namespace collisions
-            if fn_name in protected_fn_names:
-                raise AttributeError(f"{fn_name} is protected!")
-            else:
-                _classic_method = getattr(self._classic_contract.functions, fn_name)
-                _concise_method = method_class(
-                    _classic_method, self._classic_contract._return_data_normalizers
-                )
-            setattr(self, fn_name, _concise_method)
-
-    @classmethod
-    def factory(cls, *args, **kwargs):
-        return compose(cls, Contract.factory(*args, **kwargs))
+@pytest.fixture(scope="module")
+def env(gas_limit, evm_version, evm_backend, tracing, account_keys) -> BaseEnv:
+    return evm_backend(
+        gas_limit=gas_limit,
+        tracing=tracing,
+        block_number=1,
+        evm_version=evm_version,
+        account_keys=account_keys,
+    )
 
 
 @pytest.fixture
-def get_contract_from_ir(w3, optimize):
+def get_contract_from_ir(env, optimize):
     def ir_compiler(ir, *args, **kwargs):
         ir = IRnode.from_list(ir)
-        if optimize != OptimizationLevel.NONE:
+        if kwargs.pop("optimize", optimize) != OptimizationLevel.NONE:
             ir = optimizer.optimize(ir)
 
-        bytecode, _ = compile_ir.assembly_to_evm(
-            compile_ir.compile_to_assembly(ir, optimize=optimize)
-        )
+        assembly = compile_ir.compile_to_assembly(ir, optimize=optimize)
+        bytecode, _ = compile_ir.assembly_to_evm(assembly)
 
-        abi = kwargs.get("abi") or []
-        c = w3.eth.contract(abi=abi, bytecode=bytecode)
-        deploy_transaction = c.constructor()
-        tx_hash = deploy_transaction.transact()
-        address = w3.eth.get_transaction_receipt(tx_hash)["contractAddress"]
-        contract = w3.eth.contract(
-            address, abi=abi, bytecode=bytecode, ContractFactoryClass=VyperContract
-        )
-        return contract
+        abi = kwargs.pop("abi", [])
+        return env.deploy(abi, bytecode, *args, **kwargs)
 
     return ir_compiler
 
 
-def _get_contract(
-    w3,
-    source_code,
-    optimize,
-    experimental_codegen,
-    output_formats,
-    *args,
-    override_opt_level=None,
-    input_bundle=None,
-    **kwargs,
-):
-    settings = get_global_settings()
-    settings.optimize = override_opt_level or optimize
-    settings.experimental_codegen = experimental_codegen
-    out = compiler.compile_code(
-        source_code,
-        # test that all output formats can get generated
-        output_formats=output_formats,
-        settings=settings,
-        input_bundle=input_bundle,
-        show_gas_estimates=True,  # Enable gas estimates for testing
+@pytest.fixture(scope="module", autouse=True)
+def compiler_settings(optimize, experimental_codegen, evm_version, debug):
```

Comment:
what's the reason to rename this?

Reply:
I created the fixture in this branch, and in parallel another fixture was created named `global_settings`.
When I had to fix the conflict, I found this name to be much more explicit

Reply:
yea i agree, after seeing how it's used